src/scripts/mentoring-data-migration.js

```diff
@@ -0,0 +1,1325 @@
+					for (const org of orgBatch) {
+						const [, metadata] = await this.sequelize.query(
+							`
+						UPDATE ${tableName} 
+						SET ${setClauses.join(', ')}
+						FROM sessions s
+						WHERE ${whereConditions.join(' AND ')}
+						AND s.organization_id::text = '${org.org_id.replace(/'/g, "''")}'
+					`,
+							{ transaction }
+						)
+
+						totalUpdated += metadata.rowCount || 0
+					}
+				}
+
+				console.log(
+					`✅ Updated ${totalUpdated} rows in ${tableName} using session organization batching (${Math.ceil(
+						sessionOrgs.length / orgBatchSize
+					)} batches)`
+				)
+				this.stats.successfulUpdates += totalUpdated
+				await transaction.commit()
+				return
+			} else if (tableConfig.useDefaultValues) {
+				// REMOVED: Migration already sets default values, skipping this logic
+				console.log(`⚠️  Table ${tableName} uses default values - migration already handled this, skipping`)
+				await transaction.commit()
+				return
+			} else {
+				// Standard user_id lookup using user_extensions with GROUP BY organization_id approach
+				const userIdColumn = tableConfig.columns.user_id
+				const defaultTenantCode = process.env.DEFAULT_ORGANISATION_CODE || 'default'
+
+				// Get distinct organizations from user_extensions for this table
+				const [userExtByOrg] = await this.sequelize.query(`
+					SELECT DISTINCT ue.organization_id::text as org_id
+					FROM user_extensions ue
+					INNER JOIN ${tableName} t ON t.${userIdColumn} = ue.user_id
+					WHERE ue.organization_id IS NOT NULL
+					AND ue.tenant_code IS NOT NULL
+					AND ue.tenant_code != '${defaultTenantCode}'
+					ORDER BY org_id
+				`)
+
+				console.log(`🔄 Processing ${tableName} with ${userExtByOrg.length} organizations (GROUP BY approach)`)
+
+				if (userExtByOrg.length === 0) {
+					console.log(`⚠️  No organizations found for ${tableName}, skipping updates`)
+					await transaction.commit()
+					return
+				}
+
+				let totalUpdated = 0
+
+				// Process each organization in batches
+				const orgBatchSize = 5000
+				for (let i = 0; i < userExtByOrg.length; i += orgBatchSize) {
+					const orgBatch = userExtByOrg.slice(i, i + orgBatchSize)
+
+					// Process each organization individually
+					for (const org of orgBatch) {
+						// Build SET clause based on available update columns
+						const setClauses = []
+						if (tableConfig.updateColumns.includes('tenant_code')) {
+							setClauses.push('tenant_code = ue.tenant_code')
+						}
+						if (tableConfig.updateColumns.includes('organization_code')) {
+							setClauses.push('organization_code = ue.organization_code')
+						}
+						setClauses.push('updated_at = NOW()')
+
+						const [, metadata] = await this.sequelize.query(
+							`
+							UPDATE ${tableName} 
+							SET ${setClauses.join(', ')}
+							FROM user_extensions ue
+							WHERE ${tableName}.${userIdColumn} = ue.user_id
+							AND ue.organization_id::text = '${org.org_id.replace(/'/g, "''")}'
```

Comment:
@sumanvpacewisdom here you are referance is org id from table, so .replace is not require